git/filter_protocol_test.go

```diff
@@ -0,0 +1,221 @@
+package git
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type PacketReadTestCase struct {
+	In []byte
+
+	Payload []byte
+	Err     string
+}
+
+func (c *PacketReadTestCase) Assert(t *testing.T) {
+	buf := bytes.NewReader(c.In)
+	rw := newProtocolRW(buf, nil)
+
+	pkt, err := rw.readPacket()
+
+	if len(c.Payload) > 0 {
+		assert.Equal(t, c.Payload, pkt)
+	} else {
+		assert.Empty(t, pkt)
+	}
+
+	if len(c.Err) > 0 {
```

Comment:
I am just curious - why do you check 'len' here? Why don't you check against nil?